embedded/drivers/src/analog.cpp

```diff
@@ -0,0 +1,50 @@
+#include "analog.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <fstream>
+#include <iostream>
+
+#define NUM_ANALOG_PINS 7
+#define PIN_FILE_PATH "/sys/bus/iio/devices/iio:device0/in_voltage%d_raw"
+
+Analog::Analog(int pin, float scale) {
+  this->scale = scale;
+  this->pin = pin;
+}
+
+Analog::~Analog() { close(this->fd); }
+
+bool Analog::begin() {
```

Comment:
Is there a min/max to the scale factor for an analog pin? If so let's check bounds here

Reply:
i think as long as it is non-zero and positive there are no limits to the voltage divider circuit attached to the pin. I did add a check for > 0 though

Reply:
As long as it is > 0 I don't think there is a limit as this factor is determined by LV, not us. I did add a check for > 0 though